examples/crossplane/main.tf

```diff
@@ -0,0 +1,148 @@
+}
+
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.cluster.endpoint
+    token                  = data.aws_eks_cluster_auth.cluster.token
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.cluster.certificate_authority.0.data)
+  }
+}
+
+locals {
+  tenant             = "aws001"  # AWS account name or unique id for tenant
+  environment        = "preprod" # Environment area eg., preprod or prod
+  zone               = "dev"     # Environment with in one sub_tenant or business unit
+  kubernetes_version = "1.21"
+
+  vpc_cidr     = "10.0.0.0/16"
+  vpc_name     = join("-", [local.tenant, local.environment, local.zone, "vpc"])
+  cluster_name = join("-", [local.tenant, local.environment, local.zone, "eks"])
+
+  terraform_version = "Terraform v1.0.1"
+}
+
+module "aws_vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "v3.2.0"
+
+  name = local.vpc_name
+  cidr = local.vpc_cidr
+  azs  = data.aws_availability_zones.available.names
+
+  public_subnets  = [for k, v in data.aws_availability_zones.available.names : cidrsubnet(local.vpc_cidr, 8, k)]
+  private_subnets = [for k, v in data.aws_availability_zones.available.names : cidrsubnet(local.vpc_cidr, 8, k + 10)]
+
+  enable_nat_gateway   = true
+  create_igw           = true
+  enable_dns_hostnames = true
+  single_nat_gateway   = true
+
+  public_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/elb"                      = "1"
+  }
+
+  private_subnet_tags = {
+    "kubernetes.io/cluster/${local.cluster_name}" = "shared"
+    "kubernetes.io/role/internal-elb"             = "1"
+  }
+}
+#---------------------------------------------------------------
+# Example to consume aws-eks-accelerator-for-terraform module
+#---------------------------------------------------------------
+module "aws-eks-accelerator-for-terraform" {
+  source = "github.com/aws-samples/aws-eks-accelerator-for-terraform"
+
+  tenant            = local.tenant
+  environment       = local.environment
+  zone              = local.zone
+  terraform_version = local.terraform_version
+
+  # EKS Cluster VPC and Subnet mandatory config
+  vpc_id             = module.aws_vpc.vpc_id
+  private_subnet_ids = module.aws_vpc.private_subnets
+
+  # EKS CONTROL PLANE VARIABLES
+  create_eks         = true
+  kubernetes_version = local.kubernetes_version
+
+  # EKS MANAGED NODE GROUPS
+  managed_node_groups = {
+    mg_4 = {
+      node_group_name = "managed-ondemand"
+      instance_types  = ["m4.large"]
+      min_size        = "2"
+      subnet_ids      = module.aws_vpc.private_subnets
+    }
+  }
+}
+
+module "kubernetes-addons" {
+  source         = "github.com/aws-samples/aws-eks-accelerator-for-terraform//modules/kubernetes-addons"
+  eks_cluster_id = module.aws-eks-accelerator-for-terraform.eks_cluster_id
+
+  enable_crossplane = true
+
+  crossplane_helm_config = {
```

Comment:
these are default values right? Why include in the example?

Reply:
Changed the example to `enable_crossplane = true` 

However, we need make sure that users can consume our add-ons with advanced configuration.  I hope our docs are clearer for users to consume the modules. e.g., passing additional IRSA policies etc.